### PR TITLE
Upgrading to SPDK 24.05

### PR DESCRIPTION
**Pull request by dsavitskiy**

### CODE REVIEW

**Comment by dsavitskiy** [comment]
bors try

**Comment by dsavitskiy** [comment]
bors try

**Comment by dsavitskiy** [comment]
bors try

**Comment by dsavitskiy** [comment]
bors try

**Comment by dsavitskiy** [comment]
bors try

**Comment by dsavitskiy** [comment]
bors try

**Review by dsharma-dc — APPROVED** [review]

File: io-engine/src/subsys/config/opts.rs [hunk 1]
```diff
@@ -279,25 +284,29 @@ impl Default for NvmfTcpTransportOpts {
...
+                dif_insert_or_strip: o.dif_insert_or_strip,
+                reserved29: Default::default(),
+                abort_timeout_sec: o.abort_timeout_sec,
+                association_timeout: 120000,
+                transport_specific: std::ptr::null(),
+                acceptor_poll_rate: o.acceptor_poll_rate,
+                zcopy: o.zcopy,
+                reserved61: Default::default(),
+                ack_timeout: 0,
+                data_wr_pool_size: 0,
```
Comment (tiagolobocastro) [inline io-engine/src/subsys/config/opts.rs]: these are new fields, should they be added in the `NvmfTcpTransportOpts` ?

File: io-engine-tests/src/error_bdev.rs [hunk 2]
```diff
@@ -16,7 +16,13 @@ pub fn create_error_bdev(error_device: &str, backing_device: &str) {
 
     unsafe {
         // this allows us to create a bdev without its name being a uri
-        retval = create_aio_bdev(cname.as_ptr(), filename.as_ptr(), 512, false)
+        retval = create_aio_bdev(
+            cname.as_ptr(),
+            filename.as_ptr(),
+            512,
+            false,
+            false,
```
Comment (tiagolobocastro) [inline io-engine-tests/src/error_bdev.rs:24]: what is the new bool for btw?

**Review by tiagolobocastro — APPROVED** [review]

**Comment by dsavitskiy** [comment]
bors try

File: io-engine/src/subsys/config/opts.rs (hunk 1 above)
Comment (dsavitskiy) [reply io-engine/src/subsys/config/opts.rs]: Done.

File: io-engine-tests/src/error_bdev.rs (hunk 2 above)
Comment (dsavitskiy) [reply io-engine-tests/src/error_bdev.rs:24]: It's "falloc" flag.

**Comment by dsavitskiy** [comment]
bors merge

**Pull request merged**